backend/sublet/views.py

```diff
@@ -132,10 +170,35 @@ def list(self, request, *args, **kwargs):
             queryset = queryset.filter(baths=baths)
 
         # Serialize and return the queryset
-        serializer = SimpleSubletSerializer(queryset, many=True)
+        serializer = SubletSerializerSimple(queryset, many=True)
         return Response(serializer.data)
 
 
+class Images(generics.CreateAPIView):
+    serializer_class = SubletImageSerializer
+    http_method_names = ["post"]
+    permission_classes = [SubletImageOwnerPermission | IsSuperUser]
+    parser_classes = (
+        MultiPartParser,
+        FormParser,
+    )
+
+    def get_queryset(self, *args, **kwargs):
+        get_object_or_404(Sublet, id=int(self.kwargs["sublet_id"]))
+        return SubletImage.objects.filter(sublet_id=int(self.kwargs["sublet_id"]))
```

Comment:
just set sublet equal to the `get_object_or_404` and then make your query for the images filter on `sublet=sublet`, would look much cleaner.